central/networkpolicies/deployment/matcher.go

```diff
@@ -0,0 +1,131 @@
+	PolicyID string
+	Selector labels.CompiledSelector
+	Types    []storage.NetworkPolicyType
+}
+
+type policyMatcherImpl struct {
+	netpolMap map[ClusterNamespace][]selectablePolicy
+}
+
+// BuildMatcher creates a matcher with pre-loaded Network Policies from a set of ClusterNamespace filter.
+func BuildMatcher(ctx context.Context, store networkPolicyDS.DataStore, namespaceFilter set.Set[ClusterNamespace]) (Matcher, error) {
+	netpolMap, err := buildNetworkPolicies(ctx, store, namespaceFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	return &policyMatcherImpl{
+		netpolMap: netpolMap,
+	}, nil
+}
+
+func buildNetworkPolicies(ctx context.Context, store networkPolicyDS.DataStore, namespace set.Set[ClusterNamespace]) (map[ClusterNamespace][]selectablePolicy, error) {
+	result := map[ClusterNamespace][]selectablePolicy{}
+	for clusterNs := range namespace {
+		policies, err := store.GetNetworkPolicies(ctx, clusterNs.Cluster, clusterNs.Namespace)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get policies for %v", clusterNs)
+		}
+
+		for _, policy := range policies {
+			selector, err := labels.CompileSelector(policy.GetSpec().GetPodSelector())
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to create selector for policy: %s", policy.GetId())
+			}
+			result[clusterNs] = append(result[clusterNs], selectablePolicy{
+				PolicyID: policy.GetId(),
+				Selector: selector,
+				Types:    policy.GetSpec().GetPolicyTypes(),
+			})
+		}
+	}
+	return result, nil
+}
+
+// GetIsolationDetails will iterate over preloaded Network Policies and return the isolation level of
+// the resource provided as parameter.
+func (m *policyMatcherImpl) GetIsolationDetails(resource LabeledResource) IsolationDetails {
+	// Get Policies from the same cluster and namespace
+	policies, ok := m.netpolMap[ClusterNamespace{
+		Namespace: resource.GetNamespace(),
+		Cluster:   resource.GetClusterId(),
+	}]
+	if !ok {
+		// There are no policies registered for this Namespace/Cluster, stop here
+		return IsolationDetails{
+			PolicyIDs:       nil,
+			IngressIsolated: false,
+			EgressIsolated:  false,
+		}
+	}
+
+	isolationDetails := IsolationDetails{}
+	for _, policy := range policies {
+		if policy.Selector.MatchesAll() || policy.Selector.Matches(resource.GetPodLabels()) {
+			isolationDetails.PolicyIDs = append(isolationDetails.PolicyIDs, policy.PolicyID)
+			isolationDetails.IngressIsolated = isolationDetails.IngressIsolated || hasIngress(policy.Types)
+			isolationDetails.EgressIsolated = isolationDetails.EgressIsolated || hasEgress(policy.Types)
+		}
+	}
+
+	return isolationDetails
+}
+
+func hasEgress(types []storage.NetworkPolicyType) bool {
+	return hasPolicyType(types, storage.NetworkPolicyType_EGRESS_NETWORK_POLICY_TYPE)
+}
+
+func hasIngress(types []storage.NetworkPolicyType) bool {
+	if len(types) == 0 {
+		return true
+	}
+	return hasPolicyType(types, storage.NetworkPolicyType_INGRESS_NETWORK_POLICY_TYPE)
+}
```

Comment:
How about this crazy idea?

```suggestion
func hasIngress(types []storage.NetworkPolicyType) bool {
	return len(types) == 0 || hasPolicyType(types, storage.NetworkPolicyType_INGRESS_NETWORK_POLICY_TYPE)
}
```